src/context/v3.rs

```diff
@@ -0,0 +1,388 @@
+// Copyright Copyright 2024, Horizen Labs, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use super::{BoxedVC, VerifierContext, VerifierParameters};
+use crate::receipt_claim::MaybePruned;
+use crate::{
+    circuit, circuit::CircuitCoreDefV3, poseidon2_injection::Poseidon2Mix,
+    receipt::merkle::MerkleProof, receipt::succinct::SuccinctReceiptVerifierParameters,
+    receipt_claim::ReceiptClaim, segment::SegmentReceiptVerifierParameters, translate::Translate,
+    Proof,
+};
+use alloc::{boxed::Box, collections::BTreeMap, string::String};
+use risc0_binfmt_v1::{ExitCode, SystemState};
+use risc0_circuit_rv32im_v4::RV32IM_SEAL_VERSION;
+use risc0_core_v1::field::baby_bear::BabyBear;
+use risc0_zkp_v1::{
+    adapter::{ProtocolInfo, PROOF_SYSTEM_INFO},
+    core::digest::Digest,
+    verify::VerificationError,
+};
+
+impl<SC: CircuitCoreDefV3, RC: CircuitCoreDefV3> VerifierContext for V3<SC, RC> {
+    type HashSuite = HashSuiteV3;
+    type Segment = SegmentV3;
+    type Succinct = SuccinctV3;
+    fn verifier_parameters(&self) -> &VerifierParametersV3 {
+        &self.verifier_parameters
+    }
+
+    fn mut_verifier_parameters(
+        &mut self,
+    ) -> &mut VerifierParameters<Self::Segment, Self::Succinct, Self::HashSuite> {
+        &mut self.verifier_parameters
+    }
+
+    fn boxed_clone(&self) -> BoxedVC<Self> {
+        let cloned = Self {
+            verifier_parameters: self.verifier_parameters.clone(),
+            circuit: self.circuit,
+            recursive_circuit: self.recursive_circuit,
+        };
+        Box::new(cloned)
+    }
+
+    fn boxed_succinct_verifier_with_control_root(&self, control_root: Digest) -> BoxedVC<Self> {
+        Box::new(
+            V3::empty(&circuit::v3_0::CIRCUIT, &circuit::v3_0::recursive::CIRCUIT)
+                .with_suites(self.verifier_parameters.suites.clone())
+                .with_succinct_verifier_parameters(SuccinctReceiptVerifierParameters {
+                    control_root,
+                    inner_control_root: None,
+                    proof_system_info: PROOF_SYSTEM_INFO,
+                    circuit_info: self.succinct_circuit_info(),
+                }),
+        )
+    }
+
+    fn segment_circuit_info(&self) -> ProtocolInfo {
+        ProtocolInfo(*b"RV32IM:v2rev2___")
```

Comment:
This value constant is already in in the circuit module.... why we don't use the constant from the circuit. Can you try if it works? and if it works, can you check also the other contexts?